gobblin-test-harness/src/main/java/gobblin/test/setup/config/TestHarnessLauncher.java

```diff
@@ -0,0 +1,57 @@
+package gobblin.test.setup.config;
+
+import gobblin.test.execution.TestHarnessExecutor;
+
+
+/**
+ * An interface for parsing the config data for test harness. 
+ * This interface is allows to parse and validate the config setting
+ * 
+ * @author sveerama
+ *
+ * @param <I> Input folder where the config (JSON/YAML) is stored
+ * @param <C> The reader of the input for the config
+ */
+
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+* this file except in compliance with the License. You may obtain a copy of the
+* License at  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed
+* under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, either express or implied.
+*/
+public interface TestHarnessLauncher<I, C> extends ConfigStepsGenerator<C>, TestHarnessExecutor<C> {
```

Comment:
One suggestion is to actually nail down the config file instead of generifying it. The user can store in whatever format he wants and its his responsibility to feed in the information as a config which can be a simple properties object or even a hadoop config. This config can be used to execute the test harness. If the Config is an unqualified generic then, how do we plan to read the config in the framework ?